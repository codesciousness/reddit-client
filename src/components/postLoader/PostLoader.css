.PostLoader__trending {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    width: 200px;
    height: 200px;
    background: rgba(255, 255, 255, 0.7);
    border-radius: 20px;
    box-shadow: 5px 5px 5px rgba(145, 145, 165, 0.6);
    position: relative;
    margin: 2rem;
    transition: all 0.3s linear;
}

.PostLoader__trending:hover {
    transform: scale(1.2);
}

.PostLoader__trending__content {
    padding: 1rem;
    z-index: 2;
    overflow: hidden;
}

.PostLoader__trending__flair {
    display: inline-block;
    font-size: 1.25rem;
    font-weight: bold;
    text-transform: uppercase;
    margin-bottom: 1rem;
}

.PostLoader__trending__title {
    font-size: 1.25rem;
}

.PostLoader__trending__flair,
.PostLoader__trending__title {
    color:lightgray;
    background: lightgray;
    border-radius: 5px;
    opacity: 0.7;
    position: relative;
    overflow: hidden;
}

.PostLoader__trending__flair::before,
.PostLoader__trending__title::before {
    content: '';
    display: inline-block;
    position: absolute;
    height: 100%;
    width: 100%;
    background: linear-gradient(to right, transparent 0%, #E8E8E8 50%, transparent 100%);
    background-size: 300% 100%;
    border-radius: 5px;
    animation: loading 1s infinite;
}

.PostLoader__latest {
    display: flex;
    justify-content: center;
    align-items: center;
    background: rgba(255, 255, 255, 0.7);
    border-radius: 20px;
    box-shadow: 0 5px 10px rgba(145, 145, 165, 0.6);
    margin: 2rem auto;
    position: relative;
    transition: all 0.3s ease-in-out;
}

.PostLoader__latest__image {
    position: absolute;
    height: 140px;
    width: 140px;
    background: lightgray;
    border-radius: 20px;
    box-shadow: 5px 5px 10px rgba(145, 145, 165, 0.6);
    left: -64px;
    opacity: 0.9;
    transition: all 0.3s linear 0.1s;
}

.PostLoader__latest__image::before {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    background: linear-gradient(to right, transparent 0%, #E8E8E8 50%, transparent 100%);
    background-size: 300% 100%;
    border-radius: 20px;
    z-index: 1;
    animation: loading 1s infinite;
}

.PostLoader__latest__content {
    width: 75%;
    padding: 1rem;
    overflow: hidden;
}

.PostLoader__latest__flair {
    display: inline-block;
    font-size: 1.25rem;
    font-weight: bold;
    text-transform: uppercase;
    margin: 1rem auto 0.5rem auto;
}

.PostLoader__latest__title {
    font-size: 1.25rem;
    margin-bottom: 1rem;
}

.PostLoader__latest__detail {
    margin-top: 0.5rem;
}

.PostLoader__latest__detail span {
    margin: 0.5rem 2rem;
}

.PostLoader__latest__flair,
.PostLoader__latest__title,
.PostLoader__latest__author,
.PostLoader__latest__time,
.PostLoader__latest__comments{
    color:lightgray;
    background: lightgray;
    border-radius: 5px;
    opacity: 0.7;
    position: relative;
    overflow: hidden;
}

.PostLoader__latest__flair::before,
.PostLoader__latest__title::before,
.PostLoader__latest__author::before,
.PostLoader__latest__time::before,
.PostLoader__latest__comments::before {
    content: '';
    display: inline-block;
    position: absolute;
    height: 100%;
    width: 100%;
    background: linear-gradient(to right, transparent 0%, #E8E8E8 50%, transparent 100%);
    background-size: 300% 100%;
    border-radius: 5px;
    animation: loading 1s infinite;
}

@keyframes loading {
    from {
        background-position: 0 0;
    }
    to   {
        background-position: 100% 0;
    }
}

/* Media Queries */

@media only screen and (max-width: 1200px) {

    .PostLoader__latest {
        margin-top: 6rem;
    }

    .PostLoader__latest__image {
        top: -75px;
        left: 50%;
        transform: translateX(-50%);
    }

    .PostLoader__latest__content {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        flex-direction: column;
        text-align: center;
        width: 95%;
        margin-top: 3rem;
    }

    .PostLoader__latest__detail {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        flex-direction: column;
    }

    .PostLoader__latest__detail span {
        margin: 0.25rem auto;
    }
}