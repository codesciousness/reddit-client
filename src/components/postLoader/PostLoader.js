import React from 'react';
import './PostLoader.css';

const PostLoader = ({ isTrending }) => {
    if (isTrending) {
        return (
            <section className="PostLoader__trending">
                <div className="PostLoader__trending__content">
                    <h2 className="PostLoader__trending__flair">Post flair name</h2>
                    <h3 className="PostLoader__trending__title">Short title of the article...</h3>
                </div> 
            </section>
        );
    }
    else {
        return (
            <section className="PostLoader__latest">
                <div className="PostLoader__latest__image"></div>
                <div className="PostLoader__latest__content">
                    <p className="PostLoader__latest__flair">Post flair name</p>
                    <h2 className="PostLoader__latest__title">Complete title of the article</h2>
                    <p className="PostLoader__latest__detail">
                        <span className="PostLoader__latest__author">Post author</span>
                        <span className="PostLoader__latest__time">Post time</span>
                        <span className="PostLoader__latest__comments">Comment#</span>
                    </p>
                </div>
            </section>
        );
    }
}

export default PostLoader;