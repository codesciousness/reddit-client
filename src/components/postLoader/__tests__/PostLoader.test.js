import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import PostLoader from '../PostLoader';

describe('<PostLoader />', () => {
    test('PostLoader component is defined', () => {
        expect(<PostLoader />).toBeDefined();
    });
    test('matches the trending post loader snapshot', () => {
        const component = create(<PostLoader isTrending={true} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('matches the latest post loader snapshot', () => {
        const component = create(<PostLoader isTrending={false} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly when isTrending=true prop is passed', () => {
        const { container } = render(<PostLoader isTrending={true} />);
        expect(container.querySelector('.PostLoader__trending')).toBeInTheDocument();
        expect(screen.getByText('Post flair name')).toBeInTheDocument();
        expect(screen.getByText('Short title of the article...')).toBeInTheDocument();
        expect(screen.queryByText('Read more')).not.toBeInTheDocument();
        expect(screen.queryByText('Post author')).not.toBeInTheDocument();
        expect(screen.queryByText('Post time')).not.toBeInTheDocument();
    });
    test('renders correctly when isTrending=false prop is passed', () => {
        const { container } = render(<PostLoader isTrending={false} />);
        expect(container.querySelector('.PostLoader__latest')).toBeInTheDocument();
        expect(screen.getByText('Post flair name')).toBeInTheDocument();
        expect(screen.getByText('Complete title of the article')).toBeInTheDocument();
        expect(screen.getByText('Read more')).toBeInTheDocument();
        expect(screen.getByText('Post author')).toBeInTheDocument();
        expect(screen.getByText('Post time')).toBeInTheDocument();
        expect(screen.getByText('Comment#')).toBeInTheDocument();
    });
});