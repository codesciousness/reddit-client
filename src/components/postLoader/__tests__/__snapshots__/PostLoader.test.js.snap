// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostLoader /> matches the latest post loader snapshot 1`] = `
<section
  className="PostLoader__latest"
>
  <div
    className="PostLoader__latest__image"
  />
  <div
    className="PostLoader__latest__content"
  >
    <p
      className="PostLoader__latest__flair"
    >
      Post flair name
    </p>
    <h2
      className="PostLoader__latest__title"
    >
      Complete title of the article
    </h2>
    <button
      className="PostLoader__latest__button"
    >
      Read more
    </button>
    <p
      className="PostLoader__latest__detail"
    >
      <span
        className="PostLoader__latest__author"
      >
        Posted by 
        <strong>
          Post author
        </strong>
      </span>
      <span
        className="PostLoader__latest__time"
      >
        Post time
      </span>
      <span
        className="PostLoader__latest__comments"
      >
        <i
          className="far fa-comment-alt"
        />
         
        <strong>
          Comment#
        </strong>
      </span>
    </p>
  </div>
</section>
`;

exports[`<PostLoader /> matches the trending post loader snapshot 1`] = `
<section
  className="PostLoader__trending"
>
  <div
    className="PostLoader__trending__content"
  >
    <h2
      className="PostLoader__trending__flair"
    >
      Post flair name
    </h2>
    <h3
      className="PostLoader__trending__title"
    >
      Short title of the article...
    </h3>
  </div>
</section>
`;
